sched/scheduler/stateful_scheduler.go

```diff
@@ -109,6 +112,7 @@ type SchedulerConfig struct {
 	MaxRequestors           int
 	MaxJobsPerRequestor     int
 	SoftMaxSchedulableTasks int
+	HardMaxSchedulableTasks int
```

Comment:
Suggest adding to typedef comment and noting it's a suggestion vs a hard cap now (maybe rename the internal var?)